job-controller/src/main/java/feast/jobcontroller/service/JobControllerService.java

```diff
@@ -87,13 +88,17 @@ public JobCoordinatorService(
     this.specPublisher = specPublisher;
     this.groupingStrategy = groupingStrategy;
     this.featureSetSubscriptions =
-        feastProperties.getJobs().getCoordinator().getFeatureSetSelector().stream()
-            .map(JobProperties.CoordinatorProperties.FeatureSetSelector::toSubscription)
+        feastProperties.getJobs().getController().getFeatureSetSelector().stream()
+            .map(JobProperties.ControllerProperties.FeatureSetSelector::toSubscription)
             .collect(Collectors.toList());
-    this.whitelistedStores = feastProperties.getJobs().getCoordinator().getWhitelistedStores();
+    this.whitelistedStores = feastProperties.getJobs().getController().getWhitelistedStores();
     this.currentVersion = feastProperties.getVersion();
-    this.jobLabels = new HashMap<>(feastProperties.getJobs().getCoordinator().getJobSelector());
-    this.jobLabels.put(VERSION_LABEL, this.currentVersion);
+    this.jobLabels = new HashMap<>(feastProperties.getJobs().getController().getJobSelector());
+    this.jobLabels.put(VERSION_LABEL, getCurrentFeastVersion());
+  }
+
+  private String getCurrentFeastVersion() {
+    return this.currentVersion.replace(".", "-");
```

Comment:
What is happening here?

Reply:
Added comment. I need to generate value for version label that would regular expression `[\\p{Ll}\\p{Lo}\\p{N}_-]{0,63}`